### PR TITLE
Test PR for nf-core modules lint

### PR DESCRIPTION
**Pull request by drpatelh**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `<SOFTWARE>.version.txt` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: .github/workflows/nf-core-linting.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  test:
+    runs-on: ubuntu-20.04
+    name: ${{ matrix.tags }} ${{ matrix.nxf_version }}
+    needs: changes
+    if: needs.changes.outputs.modules != '[]'
+    strategy:
+      fail-fast: false
+      matrix:
+        nxf_version: ['21.03.0-edge']
```
Comment (ewels) [inline .github/workflows/nf-core-linting.yml:28]: If only planning to use one Nextflow version forever (I think this is fine?) then doesn't need to be in the matrix, can just be in the top-level `env`

File: .github/workflows/nf-core-linting.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        tags: ['${{ fromJson(needs.changes.outputs.modules) }}']
+    env:
+      NXF_ANSI_LOG: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
```
Comment (ewels) [inline .github/workflows/nf-core-linting.yml:34]: Whitespace between the steps helps readability 👌🏻 
```suggestion

      - name: Set up Python
```

File: .github/workflows/nf-core-linting.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.x"
+      - uses: actions/cache@v2
```
Comment (ewels) [inline .github/workflows/nf-core-linting.yml:38]:
```suggestion

      - uses: actions/cache@v2
```

File: .github/workflows/nf-core-linting.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
```
Comment (ewels) [inline .github/workflows/nf-core-linting.yml:43]: You're trying to load with key `${{ runner.os }}-pip-` but saving the cache with key `${{ runner.os }}-nextflow-`

File: .github/workflows/nf-core-linting.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.x"
+      - name: Install dependencies
```
Comment (ewels) [inline .github/workflows/nf-core-linting.yml:49]:
```suggestion

      - name: Install dependencies
```

File: .github/workflows/nf-core-linting.yml (hunk 1 above)
Comment (drpatelh) [reply .github/workflows/nf-core-linting.yml:28]: You read my mind! Was about to post about that on Slack too!

File: .github/workflows/nf-core-linting.yml (hunk 5 above)
Comment (drpatelh) [reply .github/workflows/nf-core-linting.yml]: Fixed https://github.com/nf-core/modules/pull/414/commits/62583a5c9fa9bedfd78d375ed57a2674d7fed6de

File: .github/workflows/nf-core-linting.yml (hunk 3 above)
Comment (drpatelh) [reply .github/workflows/nf-core-linting.yml:40]: Fixed https://github.com/nf-core/modules/pull/414/commits/62583a5c9fa9bedfd78d375ed57a2674d7fed6de

File: .github/workflows/nf-core-linting.yml (hunk 4 above)
Comment (drpatelh) [reply .github/workflows/nf-core-linting.yml:45]: No idea what this is doing to be honest.

File: .github/workflows/nf-core-linting.yml (hunk 1 above)
Comment (drpatelh) [reply .github/workflows/nf-core-linting.yml]: Ok. Fixed in  https://github.com/nf-core/modules/pull/414/commits/7902660ff663ca86b5b419dca198abed57e373eb

**Review by Emiller88 — APPROVED** [review]

**Pull request merged**